test/controllers/CheckYourAnswersControllerSpec.scala

```diff
@@ -214,83 +217,128 @@ class CheckYourAnswersControllerSpec extends SpecBase with MockitoSugar with Sum
+          val totalVatOnSales = BigDecimal(100)
+          when(salesAtVatRateService.getTotalVatOwedAfterCorrections(any())) thenReturn totalVatOnSales
+          doNothing().when(auditService).audit(any())(any(), any())
+
+          val application = applicationBuilder(userAnswers = Some(completeUserAnswers))
+            .configure("features.corrections-toggle" -> false)
             .overrides(
+              bind[VatReturnService].toInstance(vatReturnService),
               bind[VatReturnConnector].toInstance(vatReturnConnector),
-              bind[EmailService].toInstance(emailService)
-            ).build()
-
-        running(app) {
-          val request = FakeRequest(POST, routes.CheckYourAnswersController.onSubmit(period).url)
-          val result = route(app, request).value
+              bind[CorrectionConnector].toInstance(correctionConnector),
+              bind[SalesAtVatRateService].toInstance(salesAtVatRateService),
+              bind[AuditService].toInstance(auditService),
+              bind[EmailService].toInstance(emailService),
+              bind[SessionRepository].toInstance(mockSessionRepository),
+            )
+            .build()
 
-          status(result) mustEqual SEE_OTHER
-          redirectLocation(result).value mustEqual routes.ReturnSubmittedController.onPageLoad(period).url
+          running(application) {
+            val request = FakeRequest(POST, routes.CheckYourAnswersController.onSubmit(vatReturnRequest.period).url)
+            val result = route(application, request).value
+            val dataRequest = DataRequest(request, testCredentials, vrn, registration, completeUserAnswers)
+            val expectedAuditEvent = ReturnsAuditModel.build(
+              vatReturnRequest, SubmissionResult.Success, Some(vatReturn.reference), Some(vatReturn.paymentReference), dataRequest
+            )
+            val expectedAuditEventForDataEntry = ReturnForDataEntryAuditModel(vatReturnRequest, vatReturn.reference, vatReturn.paymentReference)
+
+            val userAnswersWithEmailConfirmation = completeUserAnswers.copy().set(EmailConfirmationQuery, true).success.value
+
+            status(result) mustEqual SEE_OTHER
+            redirectLocation(result).value mustEqual CheckYourAnswersPage.navigate(NormalMode, userAnswersWithEmailConfirmation).url
+
+            verify(auditService, times(1)).audit(eqTo(expectedAuditEvent))(any(), any())
+            verify(auditService, times(1)).audit(eqTo(expectedAuditEventForDataEntry))(any(), any())
+            verify(emailService, times(1))
+              .sendConfirmationEmail(eqTo(registration.contactDetails.fullName),
+                eqTo(registration.registeredCompanyName),
+                eqTo(registration.contactDetails.emailAddress),
+                eqTo(totalVatOnSales),
+                eqTo(vatReturnRequest.period)
+              )(any(), any())
+            verify(mockSessionRepository, times(1)).set(eqTo(userAnswersWithEmailConfirmation))
```

Comment:
We could verify for belt and braces that corrections connetcor is never called? your call on that one